app/models/nic/base.rb

```diff
@@ -148,6 +151,13 @@ def host_managed?
 
     protected
 
+    def compute_resource_requires_fields
+      if host.compute_resource.present?
```

Comment:
seems like it's there because this is checked on all the host's interfaces, not just the primary. putting in on the host would mean the validation needs to iterate the interfaces, which makes little sense on updates (if interface not updated, this validation should not run)